### PR TITLE
Print the file open error message when a fallback file will be attempted

### LINKED ISSUE #653: Print the file open error message when a fallback file will be attempted
Done in https://github.com/cms-sw/cmssw/pull/42615

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

In a couple of past occasions (e.g. https://github.com/root-project/root/issues/13429#issuecomment-1686079491) seeing details from the file open errors would have been useful.

Resolves https://github.com/makortel/framework/issues/653

#### PR validation:

The unit test succeeds.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
@Dr15Jones What do you think?

**Comment by Dr15Jones** [comment]
We might have to try it and see how many 'false positives' it causes and if it makes for substantially larger log files.

File: IOPool/Input/test/testFileOpenErrorExitCode.sh [hunk 1]
```diff
@@ -24,7 +24,13 @@ fi
 
 cp ${LOCAL_TEST_DIR}/sitelocalconfig/useFallbackFile/site-local-config.xml  ${SITECONFIG_PATH}/JobConfig/
 cp ${LOCAL_TEST_DIR}/sitelocalconfig/useFallbackFile/local/storage.json ${SITECONFIG_PATH}/
-cmsRun -j UseFallbackFile_jobreport.xml $F1 -- --input FileThatDoesNotExist.root && die "$F1 should have failed after file fallback but didn\'t, exit code was 0" 1
+cmsRun -j UseFallbackFile_jobreport.xml $F1 -- --input FileThatDoesNotExist.root > UseFallbackFile_output.log 2>&1 && die "$F1 should have failed after file fallback but didn't, exit code was 0" 1
+grep "Input file abc/store/FileThatDoesNotExist.root could not be opened, attempting the next fallback file" UseFallbackFile_output.log > /dev/null
```
[the remark this answers is not in the document]
Comment (makortel) [reply IOPool/Input/test/testFileOpenErrorExitCode.sh:28]: Apparently there is, thanks

**Comment by makortel** [comment]
> We might have to try it and see how many 'false positives' it causes and if it makes for substantially larger log files.

Hmm, good point. I'd think the added information is useful only when the job fails to open also all the fallback files. Maybe it would be better to limit the printout only for that case. (which then makes adding the information to the thrown exception message tempting, although it would cause the already-potentially-large exception message to become even larger...)

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
@Dr15Jones Do you have any further comments?

**Comment by Dr15Jones** [comment]
No more comments

**Comment by makortel** [comment]
+core

**Comment by perrotta** [comment]
+1

**Pull request merged**